lib/generators/json_serializable_generator.dart

```diff
@@ -2,23 +2,25 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-library source_gen.json_serial.generator;
-
 import 'dart:async';
 
 import 'package:analyzer/dart/element/element.dart';
-import 'package:analyzer/dart/element/type.dart';
-import 'package:analyzer/src/generated/utilities_dart.dart';
 import 'package:source_gen/source_gen.dart';
-import 'package:source_gen/src/annotation.dart';
+import 'package:source_gen/src/json_serializable/type_helper.dart';
+import 'package:source_gen/src/json_serializable/utils.dart';
 import 'package:source_gen/src/utils.dart';
 
 import 'json_serializable.dart';
 
+export 'package:source_gen/src/json_serializable/type_helper.dart';
+
 // TODO: toJson option to omit null/empty values
 class JsonSerializableGenerator
     extends GeneratorForAnnotation<JsonSerializable> {
-  const JsonSerializableGenerator();
+  final List<TypeHelper> typeHelpers;
+
+  const JsonSerializableGenerator(
```

Comment:
I would also add a top level `defaultHelpers` list or a `withDefaultHelpers` constructor so people don't have to hardcode the list of default ones if they supply custom ones. That way if we add more default helpers they can opt into getting those by default.